src/main/java/net/sf/jabref/model/database/DuplicationChecker.java

```diff
@@ -16,33 +17,34 @@
     // use a map instead of a set since i need to know how many of each key is in there
     private final Map<String, Integer> allKeys = new HashMap<>();
 
-    //##########################################
-    //  usage:
-    //  isDuplicate=checkForDuplicateKeyAndAdd( null, b.getKey() , issueDuplicateWarning);
-    //############################################
-    // if the newkey already exists and is not the same as oldkey it will give a warning
-    // else it will add the newkey to the to set and remove the oldkey
-    public boolean checkForDuplicateKeyAndAdd(String oldKey, String newKey) {
-        // LOGGER.debug(" checkForDuplicateKeyAndAdd [oldKey = " + oldKey + "] [newKey = " + newKey + "]");
+    /**
+     * Usage:
+     * <br>
+     * isDuplicate=checkForDuplicateKeyAndAdd( null, b.getKey() , issueDuplicateWarning);
+     *
+     * If the newkey already exists and is not the same as oldkey it will give a warning
+     * else it will add the newkey to the to set and remove the oldkey
+     *
+     * @return true, if there is a duplicate key, else false
+     */
+    public boolean checkForDuplicateKeyAndAdd(Optional<String> oldKey, Optional<String> newKey) {
```

Comment:
If I remember correctly, the caller may pass null for both oldKey and newKey. As JabRef moves away from `null` towards `Optional`, we thought, it would be a good idea to use `Optional` here, too.